src/EditorFeatures/Core/ExternalAccess/EditorConfig/Features/Hover/HoverHandler.cs

```diff
@@ -0,0 +1,183 @@
+                description = FindDescriptionForSetting(document, settingName);
+            }
+            // We are on the right of the setting and need to display the settings values description
+            else if (equalPosition != -1 && caretPosition > equalPosition)
+            {
+                var settingName = textInLine[..equalPosition].Replace(" ", "");
+                var colonPosition = textInLine.IndexOf(':');
+                var commaPosition = textInLine.IndexOf(',');
+
+                var settingValue = "";
+                if (colonPosition == -1 && commaPosition == -1)
+                {
+                    settingValue = textInLine[(equalPosition + 1)..].Replace(" ", "");
+                }
+                else if (colonPosition != -1)
+                {
+                    var values = textInLine[(equalPosition + 1)..].Split(':');
+                    var cont = equalPosition + 1;
+                    foreach (var element in values)
+                    {
+                        cont += element.Length + 1;
+                        if (caretPosition < cont)
+                        {
+                            settingValue = element.Replace(" ", "");
+                            break;
+                        }
+                    }
+                }
+                else if (commaPosition != -1)
+                {
+                    var values = textInLine[(equalPosition + 1)..].Split(',');
+                    var cont = equalPosition + 1;
+                    foreach (var element in values)
+                    {
+                        cont += element.Length + 1;
+                        if (caretPosition < cont)
+                        {
+                            settingValue = element.Replace(" ", "");
+                            break;
+                        }
+                    }
+                }
+                description = FindDescriptionForSetting(document, settingName, settingValue, displayValueInfo: true);
+            }
+
+            if (description != null)
+            {
+                return new Hover
+                {
+                    Contents = new MarkupContent
+                    {
+                        Kind = MarkupKind.Markdown,
+                        Value = description,
+                    },
+                };
+            }
+
+            return null;
+        }
+
+        private static string? FindDescriptionForSetting(TextDocument document, string settingName, string settingValue = "", bool displayValueInfo = false)
+        {
+            var workspace = document.Project.Solution.Workspace;
+            var filePath = document.FilePath;
+            var optionSet = workspace.Options;
+            Contract.ThrowIfNull(filePath);
+
+            var settingsSnapshots = SettingsHelper.GetSettingsSnapshots(workspace, filePath);
+            var codeStyleSetting = settingsSnapshots.codeStyleSnapshot?.Where(sett => sett.GetSettingName() == settingName);
```

Comment:
all of the settings here should be `IEditorConfigSettingInfo` right?  It might be easier to just combine all the different settings (codestyle, whiespace, analyzer) into one list/map and get the one you want out of that.  Then you won't have to duplicate essentially the same code for each one.